fs/readdircount_linux_amd64.go

```diff
@@ -0,0 +1,67 @@
+// +build linux,amd64
+
+package fs
+
+import (
+	"errors"
+	"os"
+	"syscall"
+	"unsafe"
+)
+
+func countDirEntries(buf []byte, n int) int {
+	count := 0
+	buf = buf[:n]
+	for len(buf) > 0 {
+		// see man page getdents(2) for struct linux_dirent64
+		reclen := *(*uint16)(unsafe.Pointer(&buf[16]))
```

Comment:
Otherwise `syscall.Getdents` wouldn't work

`Getdents` in amd64 is the 64bit version (see https://github.com/golang/go/blob/5a303aa1e96b669051a18bdc44465396a45037ea/src/syscall/syscall_linux.go#L759 , https://golang.org/src/syscall/zsyscall_linux_amd64.go?s=12082:12134#L491 and https://github.com/golang/go/blob/9e6b79a5dfb2f6fe4301ced956419a0da83bd025/src/syscall/syscall_linux_amd64.go#L9 ) and the parsing of dirents relays on the offsets of `syscall.Dirent{}` (see https://github.com/golang/go/blob/9ed0715bb66dbbd0f597f93d0bc70a3d769b1b10/src/syscall/dirent.go#L68 and https://github.com/golang/go/blob/5a303aa1e96b669051a18bdc44465396a45037ea/src/syscall/syscall_linux.go#L768 )